clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -993,7 +1314,6 @@ public void testRaiseErrorWhenNoPartitionsPendingOnDrain() throws InterruptedExc
         // We shouldn't drain batches which haven't been added to the transaction yet.
         assertTrue(drainedBatches.containsKey(node1.id()));
         assertTrue(drainedBatches.get(node1.id()).isEmpty());
-        assertTrue(transactionManager.hasFatalError());
```

Comment:
Should we be asserting something else about the `TransactionManager`? Possibly `assertFalse(transactionManager.hasFatalError())`, but there may be others.

Also, the method name says `testRaiseErrorWhen...` and there's a comment about `result in an error on drain`, but I am not seeing any errors. Do they need to be updated?